subsys/fs/Kconfig

```diff
@@ -67,6 +67,64 @@ config FS_FATFS_NUM_FILES
 config FS_FATFS_NUM_DIRS
 	int "Maximum number of opened directories"
 	default 4
+
+config FS_FATFS_LFN
+	bool "Enable long filenames (LFN)"
+	default n
+	help
+	  Without long filenames enabled, file names are limited to 8.3 format.
+	  This option increases working buffer size.
+
+config FS_FATFS_LFN_MODE
+	int "Long filename memory mode"
+	depends on FS_FATFS_LFN
```

Comment:
remove `depends on FS_FATFS_LFN` and put if around option that depends on `FS_FATFS_LFN`

Reply:
> 
> 
> remove `depends on FS_FATFS_LFN` and put if around option that depends on `FS_FATFS_LFN`

Adding if as suggested, and using choice for FS_FATFS_LFN_MODE